compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/staticcodeanalyzer/HttpServiceAnalyzer.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.http.compiler.staticcodeanalyzer;
+
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.scan.Reporter;
+import io.ballerina.stdlib.http.compiler.HttpCompilerPluginUtil;
+import io.ballerina.tools.diagnostics.Location;
+
+import static io.ballerina.stdlib.http.compiler.staticcodeanalyzer.HttpRule.AVOID_DEFAULT_RESOURCE_ACCESSOR;
+
+abstract class HttpServiceAnalyzer implements AnalysisTask<SyntaxNodeAnalysisContext> {
+    private final Reporter reporter;
+
+    public HttpServiceAnalyzer(Reporter reporter) {
+        this.reporter = reporter;
+    }
+
+    public void validateServiceMembers(NodeList<Node> members, Document document) {
+        // TODO: fix location, currently getting always -1 than expected
+        HttpCompilerPluginUtil.getResourceMethodWithDefaultAccessor(members).forEach(definition -> {
+            Location accessorLocation = definition.functionName().location();
```

Comment:
Can't we get the entire node location of the resource: `definition.location()`?
Otherwise we need to get the function definition node to get the location of the `default` keyword

Reply:
Yes, we can do that. However, we should focus on indicating the accessor location since it makes it easier for users to pinpoint the exact location where they need to make changes.

Reply:
Ack, have you tried to get it from the function definition node itself? Otherwise getting a `-1` is a bug right?

Reply:
When reporting the error, the column/line number is one less than the expected value. This is because the syntax tree location is 0-based. However, when reporting the error, this should be adjusted to a 1-based format. This needs to be fixed in the scan tool itself: https://github.com/ballerina-platform/static-code-analysis-tool/issues/29

Reply:
Ack 👍🏼